notes/stv/20200507-hadoop-k8s.txt

```diff
@@ -0,0 +1,422 @@
+spark-master            ClusterIP   10.254.252.249   <none>        7077/TCP,8080/TCP            9d
+spark-master-headless   ClusterIP   None             <none>        <none>                       9d
+
+
+# Get Persistent Storage
+# ----------------------
+kubectl get pv --sort-by=.spec.capacity.storage
+NAME                                       CAPACITY   ACCESS MODES   RECLAIM POLICY   STATUS   CLAIM                           STORAGECLASS   REASON   AGE
+pvc-e62eef27-6a06-447d-867f-13407c6e6fa7   50Gi       RWO            Delete           Bound    default/hadoop-hadoop-hdfs-nn   standard                139m
+pvc-b995c154-ddd0-4fdb-a48a-0a25b75a5f06   200Gi      RWO            Delete           Bound    default/hadoop-hadoop-hdfs-dn   standard                140m
+
+
+
+
+
+
+# ----------------------------------------------------------
+# Put some example files into HDFS
+# ----------------------------------------------------------
+# Exec into a NameNode pod, and run some HDFS commands
+#
+#
+
+kubectl exec -it -n default hadoop-hadoop-hdfs-nn-0 bash
+
+cd ~
+hdfs dfs -mkdir /hadoop/
+hdfs dfs -mkdir /hadoop/books/
+
+wget -O alice.txt https://www.gutenberg.org/files/11/11-0.txt
+wget -O holmes.txt https://www.gutenberg.org/files/1661/1661-0.txt
+wget -O frankenstein.txt https://www.gutenberg.org/files/84/84-0.txt
+
+hdfs dfs -put alice.txt holmes.txt frankenstein.txt /hadoop/books
+hdfs dfs -ls /hadoop/books
+Found 3 items
+-rw-r--r--   3 root supergroup     174481 2020-05-08 15:26 /hadoop/books/alice.txt
+-rw-r--r--   3 root supergroup     450783 2020-05-08 15:26 /hadoop/books/frankenstein.txt
+-rw-r--r--   3 root supergroup     607788 2020-05-08 15:26 /hadoop/books/holmes.txt
+
+
+# Looks like HDFS commands work successfully. How else can we validate what is going on behind the scenes?
+# Can we see these files from another hadoop node, on a different VM (K8s noed)??
+# The above pod (hadoop-hadoop-hdfs-nn-0) was on k8s-test-1-d47yw44xh3fn-node-1, let's try and list the HDFS directory we just created from another node, which has a different HDFS pod running
+
+
+# Pod hadoop-hadoop-yarn-nm-0 is running on k8s-test-1-d47yw44xh3fn-node-3, let's check that
+
+kubectl exec -it -n default hadoop-hadoop-yarn-nm-0 bash
+hdfs dfs -ls /hadoop/books
+Found 3 items
+-rw-r--r--   3 root supergroup     174481 2020-05-08 15:26 /hadoop/books/alice.txt
+-rw-r--r--   3 root supergroup     450783 2020-05-08 15:26 /hadoop/books/frankenstein.txt
+-rw-r--r--   3 root supergroup     607788 2020-05-08 15:26 /hadoop/books/holmes.txt
+
+
+# Looks like we have a working HDFS system
+
+
+# ----------------------------------------------------------
+# Check the Yarn GUI
+# ----------------------------------------------------------
+
+From local tunnel connection to the yarn ui service (10.254.142.3:8088)
+ssh -L '8088:10.254.142.3:8088' k8s-test-1-d47yw44xh3fn-master-1
+
+user@local:~$ firefox http://localhost:8088/cluster
+
+# .. Shows 2 active Nodes
+
+
```

Comment:
What is the relationship between the Spark deployment and the Hadoop deployment ?